### PR TITLE
Allow unindent without selection

### LINKED ISSUE #50279: [Text Editor] Shift+Tab shortcut only works when multiple lines are selected
### Godot version

v4.0.dev.custom_build [cb4e42155]

### System information

manjaro Gnome 21, AMD R7 200 series, Intel i3 3.0 GHZ,8GB RAM

### Issue description

When you select multiple lines and press `SHIFT+TAB` the tab goes back one step which is ok:
before:
image
after:
image

 But if the cursor is over one line only (without selecting) `SHIFT+TAB` doesnt work:
before:
image

after:
image

PS: it only works if the single line is selected, but without any selection it doesnt work!

### Steps to reproduce

* Select Multiple lines
* press the shortcut
* put the cursor at a single line without selection
* press the shortcut

### Minimal reproduction project

N/a

### PR DESCRIPTION
**Pull request by ghost**
This allows a user to press shift+tab to unindent a line without needing to make a selection. Previously, this would remove spaces or tabs preceding the cursor, and an active selection was necessary to actually unindent the line. Now, it always unindents the line the cursor is on by removing spaces or tabs at the front.

Resolves: #50279

### CODE REVIEW

File: scene/gui/code_edit.cpp [hunk 1]
```diff
@@ -603,26 +603,15 @@ void CodeEdit::do_unindent() {
...
-				spaces_to_remove = i - 1;
-				break;
-			}
-		}
-		_remove_text(cl, cc - spaces_to_remove, cl, cc);
+	} else if (line[0] == ' ') {
+		int spaces_to_remove = _calculate_spaces_till_next_left_indent(get_first_non_whitespace_column(cl));
+		_remove_text(cl, 0, cl, spaces_to_remove);
 		cursor_set_column(MAX(0, cc - spaces_to_remove));
+		return;
```
Comment (akien-mga) [inline scene/gui/code_edit.cpp:614]: The two `return`s can be removed as they're both the last statements of a branch.

**Review by Paulb23 — CHANGES_REQUESTED** [review]
This was changed on purpose. 

Might be worth allowing configuration of this behaviour? I notice that QT creator has a couple different behaviours for unindenting.

As for this specific PR, if the old behaviour is preferred by most, you can remove that entire method and call `unindent_line` directly.

**Comment by ykasehf** [comment]
Or alternatively, I could have it remove preceding spaces/tabs if there are any and otherwise unindent? That way it would cancel any tabs placed in the middle of a line while also providing the expected unindent functionality.

**Comment by Paulb23** [comment]
That's an idea worth exploring. Though I worry that it might feel inconsistent.

**Comment by akien-mga** [comment]
> This was changed on purpose.

What was the rationale for changing it? I also tend to think that the behavior proposed by this PR makes sense, at least that's what I'm used to in other editors.

The OP deleted their account so this probably needs to be redone by someone else in a new PR.

**Comment by KoBeWi** [comment]
@Calinou Try Shift+Tabbing with caret inside words.

**Pull request closed without merging**